File: DSA/src/datastructures/nodes/dual/DoubleNode.java
package datastructures.nodes.dual;

public class DoubleNode<T> {
    private T data;
    private DoubleNode<T> next;
    private DoubleNode<T> prev;

    public DoubleNode() {
        this.data = null;
        this.next = null;
        this.prev = null;
    }

    public DoubleNode(T data) {
        this.data = data;
    }

    public DoubleNode(T data, DoubleNode<T> next, DoubleNode<T> prev) {
        this.data = data;
        this.next = next;
        this.prev = prev;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public DoubleNode<T> getNext() {
        return next;
    }

    public void setNext(DoubleNode<T> next) {
        this.next = next;
    }

    public DoubleNode<T> getPrev() {
        return prev;
    }

    public void setPrev(DoubleNode<T> prev) {
        this.prev = prev;
    }
}

File: DSA/DataStructures/Nodes/DoubleNodeNS/DoubleNode.cs
namespace DSA.DataStructures.Nodes.DoubleNodeNS
{
  public class DoubleNode<T>
  {
    public T? Data { get; set; }
    public DoubleNode<T>? Next { get; set; }
    public DoubleNode<T>? Prev { get; set; }

    public DoubleNode()
    {
      this.Data = default;
      this.Next = null;
      this.Prev = null;
    }

    public DoubleNode(T data)
    {
      this.Data = data;
      this.Next = null;
      this.Prev = null;
    }

    public DoubleNode(T data, DoubleNode<T> next, DoubleNode<T> prev)
    {
      this.Data = data;
      this.Next = next;
      this.Prev = prev;
    }
  }
}